forms/session_form.go

```diff
@@ -0,0 +1,59 @@
+package forms
+
+import (
+	"google-scraper/constants"
+	"google-scraper/helpers"
+	"google-scraper/models"
+
+	"github.com/beego/beego/v2/adapter/logs"
+	"github.com/beego/beego/v2/core/validation"
+)
+
+const (
+	sessionFormErrorMessage = "Incorrect email or password"
+	sessionFormLogMessage = "Setting error on validation failed: "
+)
+
+type SessionForm struct {
+	Email    string `form:"email" valid:"Email; Required"`
+	Password string `form:"password" valid:"Required;"`
+}
+
+var currentUser *models.User
+
+func (sessionForm *SessionForm) Valid(v *validation.Validation) {
+	user, err := models.FindUserByEmail(sessionForm.Email)
+	if err != nil {
+		err := v.SetError("Email", sessionFormErrorMessage)
+		if err == nil {
```

Comment:
Not sure if this should be `err != nil` ? If yes please check another places too.

Reply:
`v.SetError("Email", sessionFormErrorMessage)` returns an error object after successfully set the error, so here I am checking `err == nil `